.semaphore/vulnix-nightly.yml

```diff
@@ -0,0 +1,65 @@
+---
+version: "v1.0"
+name: "Icepeak CI Pipeline"
+
+agent:
+  machine:
+    type: "e1-standard-2"
+    os_image: "ubuntu2004"
+
+global_job_config:
+  env_vars:
+    - name: "CRONJOB_HEALTHCHECK_URL"
+      value: "https://hc-ping.com/d80327c2-4cbc-4986-b4bb-c43002938f00"
+
+blocks:
+  - name: "Run Vulnix"
+    dependencies: []
+
+    task:
+      prologue:
+        commands:
+          - "checkout"
+
+          # Download and verify Nix installation script.
+          - curl -o install-nix-2.3.15 https://releases.nixos.org/nix/nix-2.3.15/install
+          - sha256sum --check .semaphore/install-nix-2.3.15.sha256
+
+          # Restore `/nix` cache. Create the directory first, otherwise we encounter
+          # permission errors. We do this because the Semaphore cache is faster than
+          # both Cachix and cache.nixos.org.
+          # This cache should only contain Vulnix and its dependencies
+          - "sudo mkdir /nix"
+          - "sudo chown -R semaphore:semaphore /nix"
+          - "cache restore nix-store-vulnix"
+
+          # For some reason, Semaphore CI sets this variable, but it causes the nix installation to fail
+          - unset LD_LIBRARY_PATH
+
+          # Install Nix and source the shell configuration immediately.
+          - "sh ./install-nix-2.3.15 --no-daemon"
+          - ". $HOME/.nix-profile/etc/profile.d/nix.sh"
+
+      jobs:
+        - name: "Run Vulnix on the server build derivation"
+          commands:
+            - ".semaphore/report-cronjob-start.sh"
+            - "nix-build --no-out-link ./server/default.nix"
+            - "(cd server && nix-shell --run 'vulnix -w vulnix-ignorelist.toml $(nix path-info)')"
+
+      epilogue:
+        always:
+          commands:
+            # Store a copy of the nix store. This will be refreshed daily, which
+            # is more than sufficient for this repo.
+            - "cache store nix-store-vulnix$(date -u -Idate) /nix"
```

Comment:
Is there a reason to make this job use a different Nix store cache than the regular CI runs?

Reply:
Also note that if this is required then `vulnix` should preferably be at the start of the cache key: if Semaphore can't find a cache entry with a given key it'll look for a cache entry matching the prefix until a dash, so if `nix-store-1234-56-78` doesn't exist it looks for any key matching `nix-store-1234-56-*`, then `nix-store-1234-*`, and so on.
This means that if there is no `nix-store-vulnixwhatever` entry it can also restore a regular `nix-store-whatever` entry, which might not be what you want.

Reply:
This is a relic from an experiment where I tried to run Vulnix without having to build the project. I'll update to the correct path.